### PR TITLE
Schellcheck fixes

### PR DESCRIPTION
**Pull request by SamBarker**
### Type of change

- Bugfix
- Refactoring

### Description

Fixes a bunch of shell check warnings

### Additional Context

_Why are you making this pull request?_

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Write tests
- [ ] Make sure all tests pass
- [ ] Review performance test results. Ensure that any degradations to performance numbers are understood and justified.
- [ ] Make sure all Sonarcloud warnings are addressed or are justifiably ignored.
- [ ] Update documentation
- [ ] Reference relevant issue(s) and close them after merging
- [ ] For user facing changes, update CHANGELOG.md (remember to include changes affecting the API of the test artefacts too).

### CODE REVIEW

File: kroxylicious-app/src/assembly/run-java.sh [hunk 1]
```diff
@@ -1,4 +1,5 @@
 #!/bin/sh
+# shellcheck disable=SC2039
```
Comment (robobario) [inline kroxylicious-app/src/assembly/run-java.sh:2]: might be good to add a comment next to each suppression

**Comment by SamBarker** [comment]
This spiralled some what as the upstream shell check considers a different set of shell check warnings to be errors than expected...

File: kroxylicious-app/src/assembly/run-java.sh [hunk 2]
```diff
@@ -468,10 +482,11 @@ proxy_options() {
 
 # Set process name if possible
 exec_args() {
-  EXEC_ARGS=""
   if [ -n "${JAVA_APP_NAME:-}" ]; then
     # Not all shells support the 'exec -a newname' syntax..
-    if $(exec -a test true 2>/dev/null); then
+    # shellcheck disable=SC2046
+    # disbale SC2046 as there is no word splitting to happen
```
Comment (k-wall) [inline kroxylicious-app/src/assembly/run-java.sh:488]:
```suggestion
    # disable SC2046 as there is no word splitting to happen
```

File: scripts/transition-github-release-note-state.sh [hunk 3]
```diff
@@ -52,14 +52,14 @@ if [[ ${ASSERT_NO_RELEASE_NOTES_EXIST} == "true" ]]; then
   NUM_EXIST_RELEASE_NOTES=$(curl "${CURL_ARGS[@]}" | jq --arg tag "${TAG}" '[.[] | select ( .tag_name == $tag )] | length')
   if [[ ${NUM_EXIST_RELEASE_NOTES} -gt 0 ]]; then
       >&2 echo "${NUM_EXIST_RELEASE_NOTES} release note(s) already exist for tag ${TAG}"
-   exit -1
+   exit 11
```
Comment (k-wall) [inline scripts/transition-github-release-note-state.sh:55]: why 11?

**Review by k-wall — APPROVED** [review]
Review changes and tested on Mac OS X, lgtm.
Thank you @SamBarker, that must have been tedious.

File: scripts/transition-github-release-note-state.sh (hunk 3 above)
Comment (SamBarker) [reply scripts/transition-github-release-note-state.sh:55]: Completely arbitrary on my part, it looked the most similar to `-1`. I don't believe there are any standards other than `0` being success.

Comment (SamBarker) [reply scripts/transition-github-release-note-state.sh:55]: The [Advanced Bash-Scripting Guide](https://tldp.org/LDP/abs/html/exitcodes.html) confirms my suspicions. We could use `1` as a generic error but its meaningless for debugging. I'm sticking to 11 so I can just merge this and end the pain.

**Review by k-wall — APPROVED** [review]
Rereview - lgtm

**Pull request merged**